jujugui/static/gui/src/app/app.js

```diff
@@ -847,6 +849,12 @@ YUI.add('juju-gui', function(Y) {
       if (loc.port) {
         socketUrl += ':' + loc.port;
       }
+      // If a WebSocket path is explicitly provided, it gets precedence over
+      // all the other methods to automatically calculate it.
+      var path = this.get('socket_path');
+      if (path) {
```

Comment:
it can't be undefined in reality from the call above but it's more for cosmetic.